d2/src/main/java/com/linkedin/d2/xds/XdsClientImpl.java

```diff
@@ -212,63 +212,83 @@ private void readyHandler()
     }
   }
 
-  private void handleD2NodeResponse(DiscoveryResponseData data)
+  @VisibleForTesting
+  void handleD2NodeResponse(DiscoveryResponseData data)
   {
     Map<String, NodeUpdate> updates = new HashMap<>();
     List<String> errors = new ArrayList<>();
 
-    for (Resource resource: data.getResourcesList())
+    for (Resource resource : data.getResourcesList())
     {
       String resourceName = resource.getName();
       try
       {
         XdsD2.Node d2Node = resource.getResource().unpack(XdsD2.Node.class);
+        if (d2Node != null && d2Node.getData().isEmpty())
+        {
+          _log.warn("Received a Node response with no data, resource is : {}", resourceName);
+        }
         updates.put(resourceName, new NodeUpdate(resource.getVersion(), d2Node));
-      } catch (InvalidProtocolBufferException e)
+      }
+      catch (InvalidProtocolBufferException e)
       {
         _log.warn("Failed to unpack Node response", e);
         errors.add("Failed to unpack Node response");
+        updates.put(resourceName, EMPTY_NODE_DATA);
```

Comment:
This is kind of a nit, but I think [this discussion](https://github.com/linkedin/rest.li/pull/975/files#r1518924231) should be inlined and captured like this:
```suggestion
        // Assume that the resource doesn't exist if it cannot be deserialized instead of simply ignoring it. This way
        // any call waiting on the response can be satisfied instead of timing out.
        updates.put(resourceName, new NodeUpdate(resource.getVersion(), null));
```

Reply:
Hi, @bohhyang I have talked with @PapaCharlie offline before, we thought we should not `drop the resource version` here which is may be used in the future.